src/ui/public/agg_types/metrics/moving_avg.js

```diff
@@ -23,11 +23,32 @@ import { makeNestedLabel } from './lib/make_nested_label';
 
 export const movingAvgMetricAgg = new MetricAggType({
   name: 'moving_avg',
+  dslName: 'moving_fn',
   title: 'Moving Avg',
   subtype: parentPipelineAggHelper.subtype,
   makeLabel: agg => makeNestedLabel(agg, 'moving avg'),
   params: [
-    ...parentPipelineAggHelper.params()
+    ...parentPipelineAggHelper.params(),
+    {
+      name: 'window',
+      default: 5,
+    },
+    {
+      name: 'script',
+      default: 'MovingFunctions.unweightedAvg(values)'
+    }
   ],
+  getValue(agg, bucket) {
+    /**
+     * The previous implementation using `moving_avg` did not
+     * return any bucket in case there are no documents. The `moving_fn`
+     * aggregation returns buckets with the value 0. Since our
+     * generic MetricAggType.getValue implementation would return
+     * the value 0 for null buckets, we need a specific implementation
+     * here, that preserves the null value, to be closer aligned with
+     * the previous implementation.
+     */
+    return bucket[agg.id] ? bucket[agg.id].value : 0;
```

Comment:
@timroes I don't think this is what we specified on our call.

The problem here is the following:
- the current default, 7.0, `gap_policy` of the moving function is to `skip`. This means that the function add zeros to every null value on that window, and than return the average. It's not what you wrote here, that return a zero value for buckets with an empty value. A bucket is always a average representation of N values, if the average of these N values is 0 than the bucket value is 0, if not it will be some other number. If the `gap_policy` is configured to `skip`, than it will skip the bucket calculation, returning a null bucket, that doesn't have to be confused with a 0 value average.

This is your implementation (that use the skip gap policy)
<img width="1086" alt="screen shot 2018-11-12 at 12 26 47" src="https://user-images.githubusercontent.com/1421091/48344754-43a1b280-e676-11e8-96e5-8e30666dc1e3.png">

This is the one with the line configured as we spoked: 
`return bucket[agg.id] ? bucket[agg.id].value : null;`
<img width="1084" alt="screen shot 2018-11-12 at 12 29 25" src="https://user-images.githubusercontent.com/1421091/48344888-abf09400-e676-11e8-9613-4a87aba99a02.png">

As you can see the values are the same expect that you bring down the curve on the empty bucket, instead of leaving the curve to skip that bucket and continue to the next.


This is your implementation with `inserting zeros`
<img width="1085" alt="screen shot 2018-11-12 at 12 28 37" src="https://user-images.githubusercontent.com/1421091/48344832-83689a00-e676-11e8-8708-23c491ec5124.png">



And this is my implementation with inserting zeros that are exactly the same as yours:
<img width="1088" alt="screen shot 2018-11-12 at 12 30 11" src="https://user-images.githubusercontent.com/1421091/48344912-c296eb00-e676-11e8-9096-18566bcf3893.png">